packages/gatsby-transformer-csv/README.md

```diff
@@ -78,6 +78,35 @@ the following three nodes would be created.
 ];
 ```
 
+### Illegal Characters in GraphQL
+
+Because GraphQL field names can only use alphanumeric characters and underscores gatsby-transformer-csv will replace any characters in headers not matching `/[_A-za-z0-9]/` with an underscore.  If you have spacing or punctuation that you want to preserve in your headers use the `{noheaders: true}` option, and handle your headers as data in your own code.  For example, if you have a CSV file such as:
```

Comment:
This isn't possible right now but it'd be cool if source/transformer plugins could add metadata to the connection object as then you could add the headers there and then write queries e.g.

```graphql
allIceCreamSurveyResultsCSV {
  headers
  edges {
    node {
      Type_of_Ice_Cream
      Is_it_good_
  }
}
```